adapters/httpapi_curl.c

```diff
@@ -294,6 +297,10 @@ static CURLcode ssl_ctx_callback(CURL *curl, void *ssl_ctx, void *userptr)
             result = CURLE_SSL_CERTPROBLEM;
         }
 #elif USE_MBEDTLS
+        
+        // save ssl context
+        httpHandleData->config = *((mbedtls_ssl_config*)ssl_ctx);
```

Comment:
I don't have much experience with cURL. Are all fields of the config initialized by cURL? 
mbedTLS API documentation states:

> You need to call mbedtls_ssl_config_defaults() unless you manually set all of the relevent fields yourself.

Reply:
Doing some digging, so far it is unclear if cURL calls mbedtls_ssl_config_defaults().  However it does look like this code manually sets the relevant fields(e.g., mbedtls_ssl_conf_own_cert(), mbedtls_ssl_conf_ca_chain()).  To be safe, we could call it ourselves with the same values does in tlsio_mbedtls.c.  

The memory for mbedtls_ssl_config (aka ssl_ctx) is allocated by cURL prior to sending in this callback. It looks like there is a 'curl_easy_cleanup' that is supposde to clean up memory cURL allocates, but it could be safer for us to still deallocate ourselves.

On second look, it doesn't make sense to dereference and store the ssl_ctx before these mbedtls_ssl_config (aka ssl_ctx) fields are set. But it also doesn't work to store the ssl_ctx in an 'else' clause that may not be called.  I am going to re-work the logic to reach a solution.